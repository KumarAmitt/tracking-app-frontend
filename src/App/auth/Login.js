import React, { useState } from 'react';
import { Link, Redirect } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import { getLoggedInStatus, loginUser } from '../../store/slicers/user_login';

const Login = () => {
  const dispatch = useDispatch();
  const loggedInStatus = useSelector(getLoggedInStatus);
  const [newUser, setNewUser] = useState({ username: '', password: '' });

  const handleChange = (e) => {
    setNewUser({ ...newUser, [e.target.name]: e.target.value });
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(loginUser({
      user: {
        username: newUser.username,
        password: newUser.password,
      },
    }));
  };

  if (loggedInStatus) {
    return <Redirect to="/dashboard" />;
  }

  return (
    <>
      <h2>Sign In</h2>
      <form onSubmit={handleSubmit}>
        <input type="text" name="username" placeholder="Username" value={newUser.username} onChange={handleChange} required />
        <input type="password" name="password" placeholder="Password" value={newUser.password} onChange={handleChange} required />
        <button type="submit">Login</button>
      </form>
      <br />
      <p>New User Registration form</p>
      <Link to="/registration">Sign Up</Link>
    </>
  );
};

export default Login;
