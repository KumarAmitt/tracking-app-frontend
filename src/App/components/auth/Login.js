import React, { useState } from 'react';
import { Link } from 'react-router-dom';
import { useDispatch, useSelector } from 'react-redux';
import LinearProgress from '@material-ui/core/LinearProgress';
import {
  getLoginInfo, getLoginProgress, loginUser,
} from '../../../store/slicers/userLogin';
import AppBar from '../AppBar/AppBar';
import './style/auth.css';

const Login = () => {
  const dispatch = useDispatch();
  const loginProgress = useSelector(getLoginProgress);

  const loginInfo = useSelector(getLoginInfo);
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');

  const changeUsername = (e) => {
    setUsername(e.target.value);
  };

  const changePassword = (e) => {
    setPassword(e.target.value);
  };

  const resetForm = () => {
    setUsername('');
    setPassword('');
  };

  // const handleSubmit = async (e) => {
  //   e.preventDefault();
  //   await dispatch(loginUser({
  //     user: {
  //       username,
  //       password,
  //     },
  //   }));
  //   dispatch(loadSession());
  //
  //   if (!loggedInStatus) {
  //     document.querySelector('.auth-error').classList.remove('hide');
  //   }
  //
  //   setNewUser({ ...newUser, password: '' });
  // };
  //
  // if (sessionInfo.logged_in) {
  //   return <Redirect to="/profile" />;
  // }

  const handleSubmit = (e) => {
    e.preventDefault();
    dispatch(loginUser({
      user: {
        username,
        password,
      },
    }));
    resetForm();
  };

  console.log(loginInfo);

  return (
    <>
      <AppBar title="Sign In" link="/registration" />
      <div className="auth">
        <form onSubmit={handleSubmit}>
          <input type="text" placeholder="Username" value={username} onChange={changeUsername} required />
          <input type="password" placeholder="Password" value={password} onChange={changePassword} required />
          <button type="submit">Login</button>
        </form>
        {
          loginProgress && <LinearProgress className="progress-bar" />
        }
        <div className="auth-error hide">Credentials Not Available</div>
        <div className="auth-sub">
          <p>New User Registration form</p>
          <Link to="/registration">Sign Up</Link>
        </div>
      </div>
    </>
  );
};

export default Login;
