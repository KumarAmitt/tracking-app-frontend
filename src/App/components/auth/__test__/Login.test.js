import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import Login from '../Login';
import configureAppStore from '../../../../store/configureStore';

const store = configureAppStore();

describe('Login', () => {
  it('should match the snapshot of Login component', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <Login />
        </BrowserRouter>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
