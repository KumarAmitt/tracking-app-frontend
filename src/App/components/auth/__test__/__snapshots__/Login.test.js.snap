// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login should match the snapshot of Login component 1`] = `
Array [
  <div
    className="appbar"
  >
    <a
      href="/registration"
      onClick={[Function]}
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M17.51 3.87L15.73 2.1 5.84 12l9.9 9.9 1.77-1.77L9.38 12l8.13-8.13z"
        />
      </svg>
    </a>
    <div
      className="title"
    >
      Sign In
    </div>
  </div>,
  <div
    className="auth"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        name="username"
        onChange={[Function]}
        placeholder="Username"
        required={true}
        type="text"
        value=""
      />
      <input
        name="password"
        onChange={[Function]}
        placeholder="Password"
        required={true}
        type="password"
        value=""
      />
      <button
        type="submit"
      >
        Login
      </button>
    </form>
    <div
      className="auth-error hide"
    >
      Credentials Not Available
    </div>
    <div
      className="auth-sub"
    >
      <p>
        New User Registration form
      </p>
      <a
        href="/registration"
        onClick={[Function]}
      >
        Sign Up
      </a>
    </div>
  </div>,
]
`;
